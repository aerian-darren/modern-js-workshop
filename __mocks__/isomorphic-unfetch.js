export const emoji = [
    { code: "dog", moji: "🐶", base: "dog" },
    { code: "sign of the horns", moji: "🤘", base: "sign_of_the_horns" },
    { code: "sign of the horns(p)", moji: "🤘🏼", base: "sign_of_the_horns" },
    { code: "sign of the horns(ye)", moji: "🤘🏽", base: "sign_of_the_horns" },
    { code: "sign of the horns(br)", moji: "🤘🏾", base: "sign_of_the_horns" },
    { code: "sign of the horns(bk)", moji: "🤘🏿", base: "sign_of_the_horns" },
    { code: "sign of the horns(wh)", moji: "🤘🏻", base: "sign_of_the_horns" },
    { code: "guardsman(wh)", moji: "💂🏻", base: "guardsman" },
    { code: "guardsman(p)", moji: "💂🏼", base: "guardsman" },
    { code: "guardsman(ye)", moji: "💂🏽", base: "guardsman" },
    { code: "guardsman(br)", moji: "💂🏾", base: "guardsman" },
    { code: "guardsman(bk)", moji: "💂🏿", base: "guardsman" },
    { code: "guardsman", moji: "💂", base: "guardsman" },
];

export const mockFetch = jest.fn().mockReturnValue({
    json: jest.fn().mockReturnValue(emoji),
});

export default mockFetch;

global.fetch = mockFetch;
