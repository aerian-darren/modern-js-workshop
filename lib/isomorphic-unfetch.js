export const emoji = 
    [
        { "code": "dog", "moji": "🐶", "base": "dog" }, 
        { "code": "sign of the horns", "moji": "🤘", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(p)", "moji": "🤘🏼", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(ye)", "moji": "🤘🏽", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(br)", "moji": "🤘🏾", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(bk)", "moji": "🤘🏿", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(wh)", "moji": "🤘🏻", "base": "sign_of_the_horns"}
    ]


export const mockJson = jest.fn().mockReturnValue(emoji);

export const mockFetch = jest.fn().mockReturnValue({
    json: mockJson
});

export default mockFetch;

global.fetch = mockFetch;