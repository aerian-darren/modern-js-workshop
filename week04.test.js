import "isomorphic-unfetch";
import {
  defaultMessage,
  setHeadingText,
  setUpEmojitron,
  emojitronValues,
  labelText
} from "./week04.js";

jest.mock("isomorphic-unfetch");

// eslint-disable-next-line no-unused-vars
const formTemplate = `<form id="emojitron" action="#">
    <div class="input-wrap input-select">
        <label for="emoji-select">Select your favourite emoji</label>
        <select id="emoji-select" class=""><option>Select an emoji</option>
            <option value="ðŸ¤˜">ðŸ¤˜</option>
            <option value="ðŸ’‚">ðŸ’‚</option>
            <option value="ðŸ˜‚">ðŸ˜‚</option>
            <option value="ðŸ––">ðŸ––</option>
        </select>
    </div>
</form>`;

beforeEach(() => {
  document.body.innerHTML = "<h1 id='heading'></h1>";
});

afterEach(() => {
  const form = document.getElementById("emojitron");

  if (form) form.remove();
  const h1 = document.getElementById("heading");

  if (h1) {
    h1.remove();
  }
});

describe("Emojitron creates an emoji selector with sensible defaults", () => {
  it("Has a default value in the heading", () => {
    const h1 = document.getElementById("heading");

    setHeadingText(defaultMessage);

    expect(h1.textContent).toBe(defaultMessage);
  });

  it("Has a form with id 'emojitron'", async () => {
    await setUpEmojitron();

    const emojitron = document.getElementById("emojitron");

    expect(document.forms.length).toBeGreaterThan(0);
    expect(emojitron).toBeTruthy();
    expect(emojitron).toBe(document.forms[0]);
  });

  it("Has a <select> element", async () => {
    await setUpEmojitron();

    const selectEl = document.querySelector("select");

    expect(
      selectEl.parentElement.classList.contains("uninitialised")
    ).toBeFalsy();
    expect(selectEl).toBeTruthy();
  });

  it("The select has a proper label", async () => {
    await setUpEmojitron();

    // using all selector for demonstration puroses only
    const label = document.querySelectorAll("label");

    expect(label.length).toBeGreaterThan(0);
    expect(label[0].hasAttribute("for")).toBeTruthy();
    expect(label[0].getAttribute("for")).toEqual("emoji-select");
    expect(label[0].textContent).toBe(labelText);
  });

  it("The <select> element has multiple <option> elements", async () => {
    await setUpEmojitron();

    const selectEl = document.querySelector("select");

    expect(selectEl.options.length).toBeGreaterThanOrEqual(
      emojitronValues.length
    );
  });

  it("Sets the heading when the user selects an emoji", async () => {
    await setUpEmojitron();

    const selectEl = document.querySelector("select");

    selectEl.selectedIndex = 1;

    const h1 = document.getElementById("heading");
    selectEl.dispatchEvent(new Event("change"));

    expect(h1.textContent).toEqual("ðŸ¤˜");
  });

  it("matches the snapshot", async () => {
    await setUpEmojitron();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
