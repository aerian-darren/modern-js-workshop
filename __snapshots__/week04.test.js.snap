// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emojitron creates an emoji selector with sensible defaults matches the snapshot 1`] = `"<h1 id=\\"heading\\"></h1><form id=\\"emojitron\\" name=\\"emojitron\\"><div class=\\"input-wrap input-select\\"><label for=\\"emoji-select\\">Select your favourite emoji</label><select id=\\"emoji-select\\"><option>Select an emoji</option><option value=\\"ðŸ¤˜\\">ðŸ¤˜</option>,<option value=\\"ðŸ’‚\\">ðŸ’‚</option>,<option value=\\"\\"></option>,<option value=\\"\\"></option></select></div></form>"`;
